owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/oauth/GetOIDCDiscoveryRemoteOperation.kt

```diff
@@ -0,0 +1,94 @@
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+package com.owncloud.android.lib.resources.oauth
+
+import com.owncloud.android.lib.common.OwnCloudClient
+import com.owncloud.android.lib.common.http.HttpConstants
+import com.owncloud.android.lib.common.http.methods.nonwebdav.GetMethod
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.resources.oauth.responses.OIDCDiscoveryResponse
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import timber.log.Timber
+import java.net.URL
+
+/**
+ * Get OIDC Discovery
+ *
+ * @author Abel García de Prada
+ */
+class GetOIDCDiscoveryRemoteOperation : RemoteOperation<OIDCDiscoveryResponse>() {
+
+    override fun run(client: OwnCloudClient): RemoteOperationResult<OIDCDiscoveryResponse> {
+        var result: RemoteOperationResult<OIDCDiscoveryResponse>
+
+        try {
+            val uriBuilder = client.baseUri.buildUpon().apply {
+                appendPath(WELL_KNOWN_PATH)    // avoid starting "/" in this method
+                appendPath(OPENID_CONFIGURATION_RESOURCE)
+            }.build()
+
+            val getMethod = GetMethod(URL(uriBuilder.toString())).apply {
+                addRequestHeader(OCS_API_HEADER, OCS_API_HEADER_VALUE)
+            }
+
+            val status = client.executeHttpMethod(getMethod)
+
+            val responseBody = getMethod.getResponseBodyAsString()
+
+            if (status == HttpConstants.HTTP_OK && responseBody != null) {
+                Timber.d("Successful response $responseBody")
+
+                // Parse the response
+                val moshi: Moshi = Moshi.Builder().build()
+                val jsonAdapter: JsonAdapter<OIDCDiscoveryResponse> = moshi.adapter(OIDCDiscoveryResponse::class.java)
+                val oidcDiscoveryResponse: OIDCDiscoveryResponse? = jsonAdapter.fromJson(responseBody)
+
+                result = RemoteOperationResult(RemoteOperationResult.ResultCode.OK)
+                result.data = oidcDiscoveryResponse
+
+                Timber.d("Get OIDC Discovery completed and parsed to $oidcDiscoveryResponse")
+            } else {
+                result = RemoteOperationResult(getMethod)
+                Timber.e("Failed response while getting OIDC server discovery from the server status code: $status; response message: $responseBody")
+            }
+
+        } catch (e: Exception) {
+            result = RemoteOperationResult(e)
+            Timber.e(e, "Exception while getting OIDC server discovery")
```

Comment:
```suggestion
            Timber.e(e, "Exception while getting OIDC server discovery")
            return RemoteOperationResult(e)
```

Reply:
I see your point. I will update it, thanks.